File: 149. Max Points on a Line/Solution.java
class Solution {
    public int maxPoints(int[][] points) {
        int high = 1, INF = Integer.MAX_VALUE, n = points.length;
        for (int i = 0; i < n; i++) {
            float x1 = (float) points[i][0], y1 = (float) points[i][1];
            Map<Float, Integer> counter = new HashMap<>();
            for (int j = 0; j < n; j++) {
                if (i == j) continue;
                float x2 = (float) points[j][0], y2 = (float) points[j][1];
                float dx = x1 - x2, dy = y1 - y2;
                float dydx = (dx == 0) ? INF : dy / dx;
                counter.put(dydx, counter.getOrDefault(dydx, 0) + 1);
            }
            for (Map.Entry<Float, Integer> pair : counter.entrySet()) high = Math.max(high, pair.getValue() + 1);
        }
        return high;
    }
}

File: 149. Max Points on a Line/Solution.py
class Solution:
    def maxPoints(self, points: List[List[int]]) -> int:
        high, INF, n = 1, float('inf'), len(points)
        for x1, y1 in points:
            counter = dict()
            for x2, y2 in points:
                if x1 == x2 and y1 == y2: continue
                dx, dy = x1 - x2, y1 - y2
                dydx = INF if dx == 0 else dy/dx
                if dydx not in counter: counter[dydx] = 0
                counter[dydx] += 1
            for k, v in counter.items():
                high = max(high, v + 1)
        return high